CHANGELOG.md

```diff
@@ -26,6 +26,13 @@ Released: TBD (Not before 2025-05-01)
   the type has changed to `PeggySyntaxError`, which may cause some slight need
   for rework in TypeScript-aware projects.  This was the main driver behind
   moving away from ES5. [#593](https://github.com/peggyjs/peggy/pull/593)
+- BREAKING: The grammar parser now uses your JavaScript environment's understanding
```

Comment:
I don't expect this to be a problem in practice.  The previous code was trying to do ID_Start and ID_Continue, kept to the BMP.  All of the things that used to be identifiers should still work, in any JS runtime recent enough to support ES2020.

There are 93778 characters that are out of the BMP that are valid ID_Start or ID_Continue, and they're now valid Peggy identifiers.  But note that none of them are Emoji, if you're worried about that.